pages/api/service/tests/service.test.ts

```diff
@@ -382,6 +382,136 @@ describe('Service API - GET', () => {
+      method: 'GET',
+      query: { limit, year },
+    } as any
+    const mockRes = {
+      status: jest.fn(() => mockRes),
+      json: jest.fn(),
+    } as any
+
+    await handler(mockReq, mockRes)
+
+    const response = {
+      status: mockRes.status,
+      data: mockRes.json.mock.lastCall[0].data,
+    }
+
+    expect(response.status).toHaveBeenCalledWith(200)
+
+    const received = unpopulate(
+      removeProps(response.data.map((service) => service._doc))
+    )
+    const expected = [services[3], services[4]]
+
+    expect(received).toEqual(expected)
+  })
+
+  it('should load services as DomainAdmin with limit and year', async () => {
+    await mockLoginAs(users.domainAdmin)
+    const limit = 1
+    const year = 2020
+
+    const mockReq = {
+      method: 'GET',
+      query: {limit, year},
+    } as any
+    const mockRes = {
+      status: jest.fn(() => mockRes),
+      json: jest.fn(),
+    } as any
+
+    await handler(mockReq, mockRes)
+
+    const response = {
+      status: mockRes.status,
+      data: mockRes.json.mock.lastCall[0].data,
+    }
+
+    expect(response.status).toHaveBeenCalledWith(200)
+
+    const received = unpopulate(
+      removeProps(response.data.map((service) => service._doc))
+    )
+    const expected = services.filter((service) =>
+      domains
+        .find((domain) => domain._id === service.domain)
+        .adminEmails.includes(users.domainAdmin.email) &&
+      new Date(service.date).getFullYear() === 2020
+    )
+
+    expect(received).toEqual(expected)
+  })
+
+  it('should load services as User with limit and year', async () => {
+    await mockLoginAs(users.user)
+
+    const limit = 2
+    const year = 2020
+
+    const mockReq = {
+      method: 'GET',
+      query: {year, limit},
+    } as any
+    const mockRes = {
+      status: jest.fn(() => mockRes),
+      json: jest.fn(),
+    } as any
+
+    await handler(mockReq, mockRes)
+
+    const response = {
+      status: mockRes.status,
+      data: mockRes.json.mock.lastCall[0].data,
+    }
+
+    expect(response.status).toHaveBeenCalledWith(200)
+    const received = unpopulate(
+      removeProps(response.data.map((service) => service._doc))
+    )
+    const expected = [services[3], services[4]]
+
+    expect(received).toEqual(expected)
+  })
+
+  it('should load services as GlobalAdmin with year', async () => {
+    const year = 2019
+
+    await mockLoginAs(users.globalAdmin)
+
+    const mockReq = {
+      method: 'GET',
+      query: { year },
+    } as any
+    const mockRes = {
+      status: jest.fn(() => mockRes),
+      json: jest.fn(),
+    } as any
+
+    await handler(mockReq, mockRes)
+
+    const response = {
+      status: mockRes.status,
+      data: mockRes.json.mock.lastCall[0].data,
+    }
+
+    expect(response.status).toHaveBeenCalledWith(200)
+
+    const received = unpopulate(
+      removeProps(response.data.map((service) => service._doc))
+    )
+    const expected = [services[5]]
```

Comment:
Bad approach. Test case may be broken after test-data changes. Better to handle real filtering/sorting on test-data to create proper `expected`